src/modules/ekf2/EKF/zero_velocity_update.cpp

```diff
@@ -45,15 +45,16 @@ void Ekf::controlZeroVelocityUpdate()
 
 	if (zero_velocity_update_data_ready) {
 		const bool continuing_conditions_passing = _control_status.flags.vehicle_at_rest
-				&& _control_status_prev.flags.vehicle_at_rest;
+				&& _control_status_prev.flags.vehicle_at_rest
+				&& !isVerticalVelocityAidingActive(); // otherwise the filter is "too rigid" to follow a position drift
```

Comment:
How about just skipping vz fuse below if vertical velocity aiding active? ~ line 65

Reply:
The only sources of velocity are VIO and GNSS and when active they're fusing a 3D vector so `isVerticalVelocityAidingActive` is the same as "is velocity aiding active" (maybe I should make a function that makes this obvious). I think that the same logic should apply on the horizontal axes, this is why I implemented it that way